react-nativescript-navigation/src/react-nativescript-screens/screens.tsx

```diff
@@ -442,3 +429,96 @@ export const NativeScreenContainer: React.ElementType<JSX.IntrinsicElements["fle
 
 // ScreenStack contains any number of Screen components.
 export const ScreenStack: React.ElementType<JSX.IntrinsicElements["frame"]> = "frame";
+
+type GoToScreenParams<T> = {
+  screen: React.JSXElementConstructor<T>,
+  props?: T,
+  transition?: NavigationTransition,
+  pageProps?: ScreenProps
+}
+interface NativeScreensContextData {
+  $showModal<T>(params: GoToScreenParams<T>): void
+  $navigate<T>(params: GoToScreenParams<T>): void
+  $goBack(): void
+  $closeModal(): void
+}
+
+const NativeScreensContext = React.createContext({} as NativeScreensContextData)
+
+export const NativeScreensProvider: React.FC<FrameProps> = ({ children, ...rest }) => {
+  const navRef = React.useRef<NSVElement<Frame>>()
+  function createView<T>(params: GoToScreenParams<T>, callback = (page: Page) => { }, modal?: boolean) {
+    const { screen: Screen, props } = params
+    if (!navRef.current) return
+    const ref = React.createRef<NSVElement<ContentView>>()
+    const container = new NSVElement<Page>('page')
+    if (modal) container.nativeView.background = 'transparent'
+    const key = `${Date.now()}-screen`
+    render(
+      (
+        <NativeScreensProvider {...rest}>
+          <NativeScreen {...params.pageProps} active={1} key={key}>
+            <Screen {...props ? props : undefined} />
+          </NativeScreen>
+        </NativeScreensProvider>
+      ),
+      container, () => {
+        if (params.pageProps) { 
+          Object.keys(params.pageProps).forEach((key) => {
+            if (container.nativeView[key]) {
+              container.nativeView[key] = params.pageProps[key]
+            }
+          })
+        }
+        callback(container.nativeView)
+      }, key, true)
+  }
+  function $showModal<T>(params: GoToScreenParams<T>, fullScreen?: boolean) {
+    createView(params, (page) => {
+      navRef.current.nativeView.showModal(page, {
+        animated: true,
+        transparent: true,
+        fullscreen: fullScreen,
+        context: params.props,
+        closeCallback: (_args: any) => {
+
+        }
+      });
+    })
+  }
+  function $navigate<T>(params: GoToScreenParams<T>) {
+    createView(params, (page) => {
+      navRef.current.nativeView.navigate({
+        create() {
+          return page
+        },
+        transition: params.transition,
+        backstackVisible: true,
+        context: params.props,
+        animated: true,
+      })
+    })
+  }
+
+  function $goBack() {
+    if (Frame.topmost().canGoBack()) Frame.topmost().goBack()
+  }
+
+  function $closeModal() {
+    Frame.topmost().closeModal()
```

Comment:
```suggestion
    Frame.topmost()?.closeModal();
```
(And same as above, ideally we should get a ref to the frame itself, and use `useCallback`).